drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/codec/HeaderTransport.java

```diff
@@ -171,11 +199,19 @@ public static ThriftFrame decodeFrame(ByteBuf buffer)
             // encoding info
             byte protocolId = messageHeader.readByte();
             Protocol protocol = Protocol.getProtocolByHeaderTransportId(protocolId);
-            byte numberOfTransforms = messageHeader.readByte();
-            if (numberOfTransforms > 0) {
-                // currently there are only two transforms, a cryptographic extension which is deprecated, and gzip which is too expensive
-                throw new IllegalArgumentException("Unsupported transform");
+            int numberOfTransforms = readVariableLengthInt(messageHeader);
```

Comment:
Should we validate this is some sensible number?

Reply:
It is hard to suggest a correct number. In practice the number of Transform should be a small value but protocol does not state how many there will be (the fact this is a variableLength int probably says it all :))

Reply:
But it would never be negative right

Reply:
Actually var int can be unsigned long, thus it definitely can overflow here. I will add a check.